### PR TITLE
IO.read text mode clarifications

### PR DESCRIPTION
**Pull request by XrXr**
- [DOC] Clarify that `IO.read` uses text mode
- [DOC] Mention Windows text mode EOF marker interpretation
- Test that File.read defaults to text mode

https://bugs.ruby-lang.org/issues/18882#note-13

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -14500,9 +14504,11 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
  *  Either of the following may be suffixed to any of the string read/write modes above:
  *
  *  - <tt>'t'</tt>: Text data; sets the default external encoding to +Encoding::UTF_8+;
- *    on Windows, enables conversion between EOL and CRLF.
+ *    on Windows, enables conversion between EOL and CRLF and enables interpreting +0x1A+
+ *    as an end-of-file marker.
  *  - <tt>'b'</tt>: Binary data; sets the default external encoding to +Encoding::ASCII_8BIT+;
- *    on Windows, suppresses conversion between EOL and CRLF.
+ *    on Windows, suppresses conversion between EOL and CRLF and enables interpreting +0x1A+
```
Comment (nobu) [inline io.c:14510]:
```suggestion
 *    on Windows, suppresses conversion between EOL and CRLF and disables interpreting +0x1A+
```

**Comment by nobu** [comment]
> I don't think this is super well known so it's worth mentioning as it
> can be a pitfall.

Old DOS programmers DID know it well, as a common sense 😁

File: spec/ruby/core/io/read_spec.rb [hunk 2]
```diff
@@ -104,6 +104,16 @@
...
     str.encoding.should == Encoding::ISO_8859_1
   end
+
+  platform_is :windows do
+    it "reads the file in text mode" do
+      # 0x1A is CTRL+Z and is EOF in Windows text mode.
+      content = "\x1Abbb"
+      rm_r(@fname)
+      File.binwrite(@fname, content)
+      IO.read(@fname).bytesize.should < content.bytesize
```
Comment (nobu) [inline spec/ruby/core/io/read_spec.rb:114]: Why not simply assert the result to be empty?
```suggestion
      File.binwrite(@fname, "\x1Abbb")
      IO.read(@fname).should.empty?
```

Comment (XrXr) [reply spec/ruby/core/io/read_spec.rb]: Ah that's much better. I over complicated it 😅

**Review by nobu — APPROVED** [review]

**Pull request merged**